pkg/controller/servicebindingrequest/binder.go

```diff
@@ -303,31 +304,67 @@ func (b *Binder) appendEnvVar(
 
 // appendEnvFrom based on secret name and list of EnvFromSource instances, making sure secret is
 // part of the list or appended.
-func (b *Binder) appendEnvFrom(envList []corev1.EnvFromSource, secret string) []corev1.EnvFromSource {
+func (b *Binder) appendEnvFrom(envList []corev1.EnvFromSource, binddata string) []corev1.EnvFromSource {
+
+	if b.sbr.Spec.BindingReference == nil {
+		return envList
+	}
+	dataType := b.sbr.Spec.BindingReference.ObjectType.Kind
+
 	for _, env := range envList {
-		if env.SecretRef.Name == secret {
-			b.logger.Debug("Directive 'envFrom' is already present!")
-			// secret name is already referenced
-			return envList
+		if dataType == ConfigMapKind {
+			if env.ConfigMapRef.Name == binddata {
+				b.logger.Debug("The configmap is already associated in the 'envFrom' directive.")
+				return envList
+			}
+		} else {
+			if env.SecretRef.Name == binddata {
+				b.logger.Debug("The secret is already associated in the 'envFrom' directive.")
+				// secret name is already referenced
+				return envList
+			}
 		}
 	}
 
-	b.logger.Debug("Adding 'envFrom' directive...")
-	return append(envList, corev1.EnvFromSource{
+	dataRef := corev1.EnvFromSource{
 		SecretRef: &corev1.SecretEnvSource{
 			LocalObjectReference: corev1.LocalObjectReference{
-				Name: secret,
+				Name: binddata,
 			},
 		},
-	})
+	}
+
+	if dataType == ConfigMapKind {
+		dataRef = corev1.EnvFromSource{
+			ConfigMapRef: &corev1.ConfigMapEnvSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: binddata,
+				},
+			},
+		}
+	}
+
+	b.logger.Debug("Adding 'envFrom' directive...")
+	return append(envList, dataRef)
 }
 
 // removeEnvFrom remove bind related entry from slice of "EnvFromSource".
-func (b *Binder) removeEnvFrom(envList []corev1.EnvFromSource, secret string) []corev1.EnvFromSource {
+func (b *Binder) removeEnvFrom(envList []corev1.EnvFromSource, data string) []corev1.EnvFromSource {
```

Comment:
Shouldn't this `data` be `bindData` as well(just to maintain consistency :))

Reply:
Right , done :)